hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreSetup.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Setup HiveMetastore. It does not create any database. All the tests should create a database
+ * accordingly. It should replace the existing setUp class {@link HiveMetastoreTest}
+ */
+class HiveMetastoreSetup {
```

Comment:
We can directly call static methods (start/stop) from `HiveMetastoreTest` in before(). So, no need of this class?

Reply:
It doesn't seem a good design pattern to me. `HiveMetastoreTest `  is an abstract class. Since all the tests which are extending it needs to initialise the hive catalog. Because of `BeforeAll` and `AfterAll` all the methods and members are static. 
I think it's good to use `HiveMetastoreTest` and may be a re-structring can be done with another patch.